src/gui/MainWindow.cpp

```diff
@@ -618,14 +622,18 @@ void MainWindow::setMenuActionState(DatabaseWidget::Mode mode)
     bool inDatabaseTabWidgetOrWelcomeWidget = inDatabaseTabWidget || inWelcomeWidget;
 
     m_ui->actionDatabaseMerge->setEnabled(inDatabaseTabWidget);
-
     m_ui->actionDatabaseNew->setEnabled(inDatabaseTabWidgetOrWelcomeWidget);
     m_ui->actionDatabaseOpen->setEnabled(inDatabaseTabWidgetOrWelcomeWidget);
     m_ui->menuRecentDatabases->setEnabled(inDatabaseTabWidgetOrWelcomeWidget);
     m_ui->menuImport->setEnabled(inDatabaseTabWidgetOrWelcomeWidget);
-
     m_ui->actionLockDatabases->setEnabled(m_ui->tabWidget->hasLockableDatabases());
 
+    if (m_showToolbarSeparator) {
+        m_ui->toolbarSeparator->setVisible(
+            (!inWelcomeWidget && inDatabaseTabWidget && !m_ui->tabWidget->tabBar()->isVisible())
+            || currentIndex == SettingsScreen);
+    }
```

Comment:
Should we really be "adding" this UI element, I feel like it just creates a double line that is not necessary:

![image](https://user-images.githubusercontent.com/2809491/75907887-223e7a00-5e17-11ea-8e26-37bd29284545.png)

Reply:
It's needed for consistency. The tabbar creates the same line, but when the tabbar is hidden, the line vanishes. That destroys the illusion that the toolbar area is one homogeneous area  with the tabs in front. Without the line, the top part also bleeds too much into the lower part which is particularly irritating in the light theme and on macOS where the top part is visually very distinct from the rest of the application.

Reply:
I agree it is necessary in light mode, but not in dark mode.

Reply:
I think it is needed, but it should be darker than the toolbar background, not lighter.

Reply:
I would prefer to see the line always, a clear visual border between the elements.

Reply:
Although now I think about it, the bright line makes more sense. In fact, the light theme should have a bright line as well so that the toolbar area looks sunken not raised. Otherwise the tabs couldn't be in front. That would also be more consistent with the macOS look.

Reply:
Hah, I think the separator looks good in light mode.

Reply:
I changed the swatch colour from `S_window_outline` to `S_frame_outline`, which is the same as the tabbar border for light, but slightly different for dark. Then I noticed that despite the HLine shadow being `QFrame::Plain`, there was still a subtle emboss effect that simply wasn't visible in the light theme, but very much so on a dark background.

My fix was to adjust the HLine rendering to resemble that of the tabbar base where I draw a 2px border with a specular color and then draw the actual 1px frame over that. Now it looks basically like a solid line and not like a double line anymore. It changes the overall appearance, though. The bright line colour now makes the area below the toolbar look clearly raised on Windows, while it looks sunken with a bright background.